### PR TITLE
Fix a bug in instruction cache invalidation

### LINKED ISSUE #2391: /lib/arm64/librealmreact.so Crashes on app launch
## Actual Results
[18:48] Jaideep Singh (Contractor)
    
 

Build fingerprint: 'samsung/hero2ltexx/hero2lte:8.0.0/R16NW/G935FXXU2ERGE:user/release-keys'

Revision: '9'

ABI: 'arm64'

pid: 21491, tid: 21588, name: mqt_js  >>>  <<<

signal 7 (SIGBUS), code 1 (BUS_ADRALN), fault addr 0x37bf3a90253f5

x0   0000000000000000  x1   0000000000000000  x2   0000000000000005  x3   a9037bf3a90253f5

x4   0000000000000040  x5   0000007329905788  x6   0000000000ffffff  x7   ffffffffffffffff

x8   00000000000000e2  x9   a0e2bc1e8f6365e0  x10  0000000000000001  x11  0000000000000000

x12  0000007337fbb2c8  x13  0000000000000050  x14  000000000000000d  x15  aaaaaaaaaaaaaaab

x16  000000732a38c710  x17  000000733a046cb0  x18  0000007329905a84  x19  0000007337e6f690

x20  0000007337e6f690  x21  00000073299057b8  x22  0000007329905828  x23  0000007329907588x24  000000733baecbc8  x25  0000007329907588  x26  0000000000000000  x27  0000000000000000

x28  0000000000000001  x29  0000007329905660  x30  0000007329c65e84

sp   0000007329905600  pc   00037bf3a90253f5  pstate 0000000060000000

backtrace:

#00 pc 00037bf3a90253f5  <unknown>

#01 pc 0000000000258e80 /lib/arm64/librealmreact.so

#02 pc 0000000000022414 /lib/arm64/libjscexecutor.so (_ZN8facebook3jsc10JSCRuntimeC2Ev+28)

#03 pc 00000000000254d0  /lib/arm64/libjscexecutor.so (_ZN8facebook3jsc14makeJSCRuntimeEv+28)

#04 pc 0000000000018b58  /lib/arm64/libjscexecutor.so

#05 pc 00000000000a4468  /lib/arm64/libreactnativejni.so (_ZN8facebook5react16NativeToJsBridgeC2EPNS0_17JSExecutorFactoryENSt6__ndk110shared_ptrINS0_14ModuleRegistryEEENS5_INS0_18MessageQueueThreadEEENS5_INS0_16InstanceCallbackEEE+252)

#06 pc 000000000009c0e0  /lib/arm64/libreactnativejni.so

#07 pc 0000000000065d7c  ==/lib/arm64/libreactnativejni.so

#08 pc 0000000000063cac  /lib/arm64/libreactnativejni.so

#09 pc 0000000000059288  /lib/arm64/libreactnativejni.so (_ZN8facebook3jni6detail13MethodWrapperIMNS_5react15JNativeRunnableEFvvEXadL_ZNS4_3runEvEES4_vJEE8dispatchENS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassIS4_NS3_8RunnableEE8JavaPartESB_vE11_javaobjectEEE+32)

#10 pc 0000000000059204 /lib/arm64/libreactnativejni.so (_ZN8facebook3jni6detail15FunctionWrapperIPFvNS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassINS_5react15JNativeRunnableENS6_8RunnableEE8JavaPartES8_vE11_javaobjectEEEEXadL_ZNS1_13MethodWrapperIMS7_FvvEXadL_ZNS7_3runEvEES7_vJEE8dispatchESE_EESD_vJEE4callEP7_JNIEnvP8_jobject+56)
#11 pc 000000000007a63c /oat/arm64/base.odex (offset 0x75000)

## Steps to Reproduce
Launching the app

## Code Sample
defaultConfig {
        applicationId "package-name"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 137
        versionName "8.2"
        multiDexEnabled true
        ndk {
            abiFilters 'armeabi-v7a','arm64-v8a','x86','x86_64' // Added
        }
        vectorDrawables.useSupportLibrary = true
        resValue "string", "build_config_package", "package-name"
        if (BUILD_FOR_APPCENTER=='false') {
            signingConfig signingConfigs.release
        }
    }

## Version of Realm and Tooling

- Realm JS SDK Version: 2.28.0
- Node or React Native: ^0.59.4
- Client OS & Version: Android 8
- Which debugger for React Native: ?/None

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #2391 

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 🚦 Tests~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: src/android/jsc_override.cpp [hunk 1]
```diff
@@ -143,6 +145,7 @@ static JSGlobalContextRef create_context(JSContextGroupRef group, JSClassRef glo
 
 static void swap_function()
 {
+    __android_log_print(ANDROID_LOG_INFO, "svap_function", "BEGIN");
```
Comment (cmelchior) [inline src/android/jsc_override.cpp]: Maybe just log as debug? Same for others

Comment (kneth) [reply src/android/jsc_override.cpp]: I should have been explicit about it: I plan to remove all logging before releasing.

File: src/android/jsc_override.cpp [hunk 2]
```diff
@@ -143,6 +145,7 @@ static JSGlobalContextRef create_context(JSContextGroupRef group, JSClassRef glo
...
     static int8_t s_orig_code[HOOK_SIZE];
```
Comment (nhachicha) [inline src/android/jsc_override.cpp]: don't we need the bytes alignment `static alignas(16) s_orig_code[HOOK_SIZE];` ?

**Comment by diegolmello** [comment]
@kneth Do you know if this is similar or may help fix this RN issue? https://github.com/facebook/react-native/issues/24261

File: src/android/jsc_override.cpp (hunk 2 above)
Comment (kneth) [reply src/android/jsc_override.cpp]: Not sure, but I leave a TODO comment about it.

**Pull request merged**